sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -87,6 +87,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: SQLConf)
       CollapseRepartition,
       CollapseProject,
       CollapseWindow,
+      CollapseConcat,
```

Comment:
This is not part of `Operator combine`. Maybe move it to the spot around `SimplifyCasts`